File: RPi Experiments/RPi2-1/C/07_potBlinkingLed.c
#include<wiringPi.h>
#include<stdio.h>
#include<pcf8591.h>

#define PCF 120 //base pin for pcf8591
#define ledPin 0 //wiringPi pin number for the LED

int main(void)
{
    int value;
    int dc;

    if(wiringPiSetup() == -1)
	{
		printf("Setup wiring Pi failed!");
		return 1;
    };
	
	pinMode(ledPin, OUTPUT); //initialize the LED pin as output
    
    pcf8591Setup(PCF, 0x48); //setup pcf8591 on address 0x48

    while(1)
    {
        value = analogRead(PCF + 0); //read the value from potentiometer
        printf("Milliseconds: %d\n", value);
        analogWrite(PCF + 0, value);
        digitalWrite(ledPin,HIGH); //turn on the LED
		printf("Led ON...\n");
		delay(value); //wait for the number of ms in value
		digitalWrite(ledPin,LOW); //turn off the LED
		printf("Led OFF...\n"); 
		delay(value); //wait for the number of ms in value
    }

    return 0;
}

File: RPi Experiments/RPi2-1/Python/07_potBlinkingLed.py
import RPi.GPIO as GPIO
import time
import PCF8591 as ADC

ledPin = 11 # physical pin number for the LED

def setup():
    GPIO.setwarnings(False) # don't show warnings if GPIO is in use
    GPIO.setmode(GPIO.BOARD) # use GPIO by physical location
    GPIO.setup(ledPin,GPIO.OUT) # initialize the LED pin as output
    ADC.setup(0x48)

def loop():
    while True:
        value = ADC.read(0) # read the potentiometer value
        ms = float(value) / 1000
        print 'Milliseconds: ' + str(ms)
        ADC.write(ms)
        print "Led ON..."
        GPIO.output(ledPin,GPIO.LOW) # turn on led
        time.sleep(ms) # wait for the number of seconds in ms
        print "Led OFF..."
        GPIO.output(ledPin,GPIO.HIGH) # turn off led
        time.sleep(ms) # wait for the number of seconds in ms

def destroy():
    ADC.write(0)
    GPIO.output(ledPin,GPIO.HIGH) # turn off led
    GPIO.cleanup() # release resource

if __name__ == '__main__': # program start from here
    try:
        setup()
        loop()

    except KeyboardInterrupt: # when 'Ctrl+C' is pressed, the child program destroy() will be  executed.
        destroy()